ostd/src/mm/heap_allocator/mod.rs

```diff
@@ -93,23 +96,28 @@ impl LockedHeapWithRescue {
             size / PAGE_SIZE
         };
 
-        let allocation_start = {
-            let mut page_allocator = PAGE_ALLOCATOR.get().unwrap().lock();
+        let allocated_cont_pages = {
             if num_frames >= MIN_NUM_FRAMES {
-                page_allocator.alloc(num_frames).ok_or(Error::NoMemory)?
+                alloc_contiguous(num_frames * PAGE_SIZE, |_| HeapMeta::default())
```

Comment:
> Could you explain the motivation of this PR? I can't see any benefit of acquiring `ContPages` and then forget them immediately. From this PR alone, it looks that we are just paying the extra cost of `ContPages::from_unused` without getting anything in return.

This PR aims not to enhance performance but to fix the missing meta management in Heap Memory. 

Currently, almost all the allocated memory is managed by `meta`, whose allocation correctness can be checked by `from_free` but not only the page allocator itself. The double-checked mechanism makes the allocation process more robust, but the previous implementation did not include heap memory in the `meta`.

Since the future page allocator and its correlated utilities rely heavily on `meta`, I suggest aligning heap memory allocation with the `meta` design. The `ContPages::from_unused` is the corresponding operation while managing allocated heap pages with `meta`.